gcloud-java-resourcemanager/src/main/java/com/google/gcloud/spi/ResourceManagerRpc.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.spi;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.cloudresourcemanager.model.Policy;
+import com.google.api.services.cloudresourcemanager.model.Project;
+import com.google.common.collect.ImmutableList;
+import com.google.gcloud.resourcemanager.ResourceManagerException;
+
+import java.util.Collections;
+import java.util.List;
+
+public interface ResourceManagerRpc {
+
+  public enum Permission {
+    CREATE,
+    DELETE,
+    GET,
+    LIST,
+    UPDATE,
+    GET_IAM_POLICY,
+    SET_IAM_POLICY
+  }
+
+  public enum DeleteResult {
+    SUCCESS,
+    ALREADY_DELETED
+  }
+
+  public enum UndeleteResult {
+    SUCCESS,
+    DELETE_IN_PROGRESS,
+    GONE
+  }
+
+  public enum SetIamPolicyResult {
+    SUCCESS,
+    ETAG_NOT_MATCH
```

Comment:
Oh, sorry I just read your comment about it:

> We could probably add a return type for SetIamPolicy as well. It can fail to set the policy if the user chooses to supply an etag and that etag doesn't match the server policy's etag.

That is more clear now, so if a user does not supply an etag there will be no check.
If a user does supply an etag, then the set will verify that the current policy for that project
has a matching etag.

So, if I don't care about the etag and update the project with a new Policy, that Policy will
get a new etag?